src/core/filters/meta/filter.rs

```diff
@@ -0,0 +1,470 @@
+        };
+
+        top.sz = if lmo.is_byte() {
+            1
+        } else if lmo.is_short() {
+            2
+        } else if lmo.is_int() {
+            4
+        } else if lmo.is_long() {
+            8
+        } else {
+            bail!("unexpected numeric type");
+        };
+    }
+
+    top.cmp = cmp_op as u8;
+
+    Ok(op)
+}
+
+#[derive(Clone)]
+pub(crate) struct FilterMeta(pub(crate) Vec<MetaOp>);
+
+impl FilterMeta {
+    pub(crate) fn from_string(fstring: String) -> Result<Self> {
+        let btf = &inspector()?.kernel.btf;
+        let mut ops: Vec<_> = Vec::new();
+        let mut offt: u32 = 0;
+        let mut stored_offset: u32 = 0;
+
+        // Check the correctness, perform preliminar checks for the
+        // operator type against the target (e.g. >= against number).
+        //
+        // Once the lmo type is known, compare the rval against the
+        // lmo type (e.g. INT against LONG, sign)
+        let re = Regex::new(MF_RE)?;
+
+        let Some((_, [lval, op, rval])) = re.captures(&fstring).map(|caps| caps.extract()) else {
+            bail!("Invalid filter expression.")
+        };
+
+        let mut fields: Vec<_> = lval.split('.').collect();
+
+        // The captures ensure at least two elements are present
+        let init_sym = fields.remove(0);
+
+        if !init_sym.eq("sk_buff") {
+            bail!("unsupported data structure {init_sym}. sk_buff must be used.")
+        }
+
+        let (btf, mut r#type) = btf
+            .resolve_type_by_name(init_sym)
+            .map_err(|e| anyhow!("unable to resolve sk_buff data type {e}"))?;
+
+        for (pos, field) in fields.iter().enumerate() {
+            let sub_node = walk_btf_node(btf, &r#type, field, offt);
+            match sub_node {
+                Some((offset, snode)) => {
+                    if pos < fields.len() - 1 {
+                        // Type::Ptr needs indirect actions (Load *Ptr).
+                        //   Offset need to be reset
+                        // Named Structs or Union return (level matched) but are
+                        //   still part of the parent Struct, so the offset has to
+                        //   be preserved.
+
+                        let (ind, x) = next_walkable(btf, snode)?;
+                        let one = 1;
+
+                        match ind.cmp(&one) {
+                            std::cmp::Ordering::Equal => {
+                                offt = 0;
+                                // Emit load Ptr
+                                let mut op: MetaOp = unsafe { std::mem::zeroed::<_>() };
+                                op.l.offt = u16::try_from(offset / 8)?;
+                                op.l.r#type = PTR_BIT;
+                                ops.push(op);
+                            }
+                            std::cmp::Ordering::Greater => {
+                                bail!("multiple pointers are not supported")
```

Comment:
Uhm, I mean, I guess it's fine like it is, but ok.

Reply:
> It might be just me, when reading this I initially thought `sk_buff.ptr1.ptr2` wasn't supported. But IIUC this is about `** ptr`?

your understanding is spot on.
Thanks for the feedback, I didn't realize this could be misleading.
Let's go with that, then.